src/optimize.c

```diff
@@ -1175,12 +1176,17 @@ Expression *Expression_optimize(Expression *e, int result, bool keepLvalue)
             }
             if (e->e1->isBool(true))
             {
-                // Replace with (e1, 1)
-                ret = new CommaExp(e->loc, e->e1, new IntegerExp(e->loc, 1, e->type));
-                ret->type = e->type;
+                // Replace with true
+                ret = new IntegerExp(e->loc, 1, Type::tbool);
+                if (e->type->toBasetype()->ty == Tvoid)
+                {
+                    ret = new CastExp(e->loc, ret, Type::tvoid);
+                    ret->type = e->type;
+                }
```

Comment:
What if e1 has side effects at runtime, but during ctfe (can happen with ``if (__ctfe)``) ?

I think the CommaExp should remain here.